src/test/java/de/tum/in/www1/artemis/exercise/quizexercise/QuizSubmissionIntegrationTest.java

```diff
@@ -407,11 +416,22 @@ void testQuizSubmitEmptyQuizInLiveMode() throws Exception {
     @WithMockUser(username = TEST_PREFIX + "student7", roles = "USER")
     @EnumSource(QuizMode.class)
     void testQuizSubmitNoDatabaseRequests(QuizMode quizMode) throws Exception {
+        CountDownLatch lock = new CountDownLatch(1);
+
+        var topic = hazelcastInstance.getTopic(HAZELCAST_CACHED_EXERCISE_UPDATE_TOPIC);
+        topic.addMessageListener(o -> lock.countDown());
+
         Course course = courseUtilService.createCourse();
         QuizExercise quizExercise = quizExerciseUtilService.createQuiz(course, ZonedDateTime.now().minusHours(5), null, quizMode);
         quizExercise.setDuration(360);
         quizExercise.getQuizBatches().forEach(batch -> batch.setStartTime(ZonedDateTime.now().minusMinutes(5)));
-        quizExerciseService.save(quizExercise);
+        quizExercise = quizExerciseService.save(quizExercise); // <- responsible for saving the exercise into the distributed exercise cache
+        final long exerciseId = quizExercise.getId();
+
+        // Wait until the exercise update got processed
+        if (!lock.await(2000, TimeUnit.MILLISECONDS)) {
+            fail("Timed out waiting for the quiz exercise cache.");
+        }
```

Comment:
This looks OK, but might still be kind of fragile, depending on the execution order of the listeners (I could not find information on that)